src/NLog/Config/XmlLoggingConfiguration.cs

```diff
@@ -272,7 +272,10 @@ public override IEnumerable<string> FileNamesToWatch
         public override LoggingConfiguration Reload()
         {
             if (!string.IsNullOrEmpty(_originalFileName))
-                return new XmlLoggingConfiguration(_originalFileName, LogFactory);
+            {
+                LogFactory.LoadConfiguration(_originalFileName);
```

Comment:
FYI, needed this otherwise IAppEnvironment will be ignored

Reply:
Breaking change as the Reload now assigns the Configuration. What happens if the LogFactory.LoadConfiguration fails?

Reply:
> Breaking change as the Reload now assigns the Configuration

Well I always was under the impression that was a bug. Or do you think it was useful to have that separated? 

> No longer possible to call explicit Reload() ,and then adjust the returned config and then assign the config.

Why is that not possible now?

Reply:
Because `LogFactory.LoadConfiguration` will assign the configuration, so any adjustments are too late.

Reply:
But I can certainly agree that the config-reload is full of pitfalls. But not sure if the fix should be implemented on XmlLoggingConfiguration directly.

Reply:
> But maybe the improvement can be implemented without changing XmlLoggingConfiguration.Reload() ?

Good idea, will look into it. Will be tomorrow I guess

Reply:
Fixed, please double check 

PS I would appreciate an Approve  "flag"  :)

Reply:
Have created #3941 that improves the mocking of `XmlReader.Create` for `XmlLoggingConfiguration`.